src/pages/Lend/LendByCategoryPage.vue

```diff
@@ -225,6 +237,84 @@ export default {
 			// 		},
 			// 	});
 			// }
+		},
+		fetchRemainingLoanChannels() {
+			const ssrLoanIds = [];
+			// pick loan ids from each
+			_each(this.categories, category => {
+				_each(category.loans.values, loan => {
+					ssrLoanIds.push(loan.id);
+				});
+			});
+			// Client Fetch the remaining category rows
+			return this.apollo.query({
+				query: loanChannelQuery,
+				variables: {
+					ids: _drop(this.realCategoryIds, ssrRowLimiter),
+					excludeIds: ssrLoanIds,
+					// @todo variables for fetching data for custom channels
+				},
+			}).then(({ data }) => {
+				// add our remaining loan channels
+				this.clientCategories = _get(data, 'lend.loanChannelsById') || [];
+				Promise.resolve();
+			});
+		},
+		fetchRecentlyViewed() {
+			// Setup Recently Viewed Loans data for inclusion in page load loan row analytics
+			// Read assignment for Recently Viewed Loans EXP
+			const recentlyViewedEXP = this.apollo.readQuery({
+				query: experimentQuery,
+				variables: { id: 'recently_viewed_loans' }
+			});
+			this.showRecentlyViewed = _get(recentlyViewedEXP, 'experiment.version') === 'variant-a';
+			// if Recently Viewed Exp is active look for loans in local storage
+			if (this.showRecentlyViewed) {
+				// fetch recently viewed from localStorage (currently set in wwwApp on Borrower Profile)
+				const recentlyViewed = WebStorage('recentlyViewedLoans');
+				// decode, parse then set recently viewed loan data
+				try {
+					this.recentLoanIds = JSON.parse(atob(recentlyViewed));
+				} catch (e) {
+					// no-op
+				}
+			}
+		},
+		activateWatchers() {
+			// Create an observer for changes to the categories (and their loans)
+			const categoryObserver = this.apollo.watchQuery({
+				query: loanChannelQuery,
+				variables: {
+					ids: this.realCategoryIds,
+					// @todo variables for fetching data for custom channels
+				},
+			});
+
+			// Watch for and react to changes to the query
+			this.apollo.watchQuery({ query: lendByCategoryQuery }).subscribe({
+				next: ({ data }) => {
+					this.setRows(data);
+					this.isAdmin = !!_get(data, 'my.isAdmin');
+					this.itemsInBasket = _map(_get(data, 'shop.basket.items.values'), 'id');
+					// Update the categories observer with the new setting, triggering updates
+					categoryObserver.setVariables({
+						ids: this.realCategoryIds,
+						// @todo variables for fetching data for custom channels
+					});
+				},
+			});
+
+			// React to changes to the category data
```

Comment:
This seems unnecessary and I couldn't figure out what was enabled with it in place.

I did find this was triggering an additional full query for all loan data that was redundant.

Reply:
You can drop categoryObserver entirely. It enabled admins to immediately see the changes they made to which categories were shown on the page. Definitely not essential, as they can always just refresh the page.

The reason is triggers another network call is because of how the loading of categories is split between the server and the client now. The queries in the apollo cache are keyed based on their arguments as well, so the results of two different api calls using the same query but different arguments will both be cached. This observer is setup ti watch the results of a query for *all* the categories, but now that the prefetching is split, the cache has a query for some categories (from the server) and a second query for some others (from the client). Since the cache doesn't have the results for a single query that fetched all the categories, creating this observer triggers a full api request. 

Since this isn't an important feature, I think we can just get rid of it. If we wanted to continue supporting this feature, we could maybe wait to create the observer until after an admin makes changes.

btw, this wouldn't have affected ssr load times because any promises or observers made in the created() callback (or anywhere in the vue app) never resolve on the server, as the app is rendered synchronously. Prefetching (which is obviously async) only works because it is handled before the app is given to the vue server renderer.

Reply:
Thanks for the info that makes sense about caching the initial queries and not a single unified query.

Also the ssr statement mostly makes sense... I do wonder if it would still try though.